### ISSUE TITLE
[closures] compile crash; local variable in anonymous function 

### ISSUE THREAD

**Issue by valipan** [vlang/v#4780]
**V version:**  0.1.27 8866773
**OS:** linux amd64

**What did you do?**
```
fn main(){
	a:= 1
	
	func:= fn(){
		println('a=$a')
	}

	func()
}
```

**What did you expect to see?**
I expected the program to compile successfully.
 
**What did you see instead?**
```
==================
/home/dev/.cache/v/anon1.tmp.c: In function ‘anon_71__1’:
/home/dev/.cache/v/anon1.tmp.c:1127:33: error: ‘a’ undeclared (first use in this function)
  println(_STR("a=%"PRId32"", 1, a));
                                 ^
/home/dev/.cache/v/anon1.tmp.c:1127:33: note: each undeclared identifier is reported only once for each function it appears in
/home/dev/.cache/v/anon1.tmp.c: In function ‘_STR’:
/home/dev/.cache/v/anon1.tmp.c:1240:14: warning: pointer targets in passing argument 1 of ‘tos2’ differ in signedness [-Wpointer-sign]
  return tos2(buf);
              ^~~
/home/dev/.cache/v/anon1.tmp.c:999:8: note: expected ‘byteptr {aka unsigned char *}’ but argument is of type ‘char *’
 string tos2(byteptr s);
        ^~~~
...
==================
```

**Comment by nedpals** [comment]
Access to parent scoping has not been implemented :)

**Comment by sheerun** [comment]
I don't think it should be implemented for performance. But checker should not allow it.

**Comment by Jonatino** [comment]
Another example of where this would be nice to have
```
fn load_item_spawns() {
	mut spawns := []ItemSpawn{}

    parse_spawn := fn(file string)  {
       	data := os.read_file(file) or {
			panic('error reading file $file')
			return
		}
		mut current_spawns := json.decode([]ItemSpawn, data) or {
			eprintln('Failed to parse item spawns')
			return
		}
		spawns << current_spawns
    }

	os.walk("./data/items/spawns/", parse_spawn)

	println(spawns.len)
}
```

Instead I have to duplicate the os.walk code in order to get the previous code to work.

**Comment by fxn** [comment]
I believe this one can be closed.

In the current version (`V 0.2.2 a9fb24c`), the program does NOT compile, and this is correct according to the current documentation:

```
% v foo.v    
foo.v:2:2: warning: unused variable: `a`
    1 | fn main() {
    2 |     a:= 1
      |     ^
    3 | 
    4 |     func:= fn () {
foo.v:5:15: error: undefined ident: `a`
    3 | 
    4 |     func:= fn () {
    5 |         println('a=$a')
      |                     ^
    6 |     }
    7 |
```

If you pass the variable to the function, as documented today:

```v
func:= fn [a] () {
	println('a=$a')
}
```


the example runs:

```
% v run foo.v
a=1
```
